src/systems/physics/Physics.cc

```diff
@@ -722,25 +825,27 @@ void PhysicsPrivate::CreatePhysicsEntities(const EntityComponentManager &_ecm)
           return true;
         }
 
-        auto childLinkPhysIt =
-            this->entityLinkMap.find(_jointInfo->Data().childLink);
-        if (childLinkPhysIt == this->entityLinkMap.end())
+        auto childLinkDetachableJointFeature = entityCast(
+            _jointInfo->Data().childLink, this->entityLinkMap,
+            this->entityLinkDetachableJointMap);
+        if (!childLinkDetachableJointFeature)
         {
-          ignwarn << "DetachableJoint's child link entity ["
-                  << _jointInfo->Data().childLink << "] not found in link map."
-                  << std::endl;
-          return true;
+          ignwarn << "Can't process DetachableJoint component, physics engine "
+                  << "missing AttachFixedJointFeature" << std::endl;
+
+          // Break Each call since no DetachableJoints can be processed
+          return false;
```

Comment:
By returning early, we are saying that the only way `entityCast` returns a `nullptr` is if the requested feature is missing, but looking at the code in `entityCast`, it can also return a `nullptr` if the entity is not in the `minimumMap`. I'm not sure, at least in this case, if the code will get to calling `entityCast` with the entity not being in `minimumMap`, but the check is in there nonetheless. 

This got me thinking...do we need to pass in the `minimumMap` at all? Can we not just pass the `EntityPtr` found from the the `minimumMap`? Looking at most places where `entityCast` is used, the code surrounding it already does a lookup in the `minimumMap`, so the extra check in `entityCast`  can be avoided by passing in the `EntityPtr`.

Reply:
Yeah you're right, most of the time the lookup has already happened. I think the first couple that I implemented didn't need the external lookup and that's why I moved it into the cast. Let me move it out.

Reply:
I guess it was a tie, 3 needed map lookup while 3 didn't: 1ce537a